Library/Homebrew/cask/audit.rb

```diff
@@ -787,13 +795,33 @@ def cask_plist_min_os
           next unless File.exist?(plist_path)
 
           plist = system_command!("plutil", args: ["-convert", "xml1", "-o", "-", plist_path]).plist
-          plist_min_os = plist["LSMinimumSystemVersion"].presence
-          break if plist_min_os
+          min_os = plist["LSMinimumSystemVersion"].presence
+          break if min_os
+
+          next unless (main_binary = get_plist_main_binary(path))
+          next if !File.exist?(main_binary) || File.open(main_binary, "rb") { |f| f.read(2) == "#!" }
+
+          macho = MachO.open(main_binary)
```

Comment:
Might also be nicer to add methods to `MachOShim` so can handle `Pathname`s instead of `MachO` objects directly.